src/users/users.service.ts

```diff
@@ -0,0 +1,76 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { CreateUserDto, LoginUserDto } from '../dto/users-dto';
+import { Login } from '../model/users/login.entity';
+import { User } from '../model/users/user.entity';
+import { Repository } from 'typeorm';
+import * as bcrypt from 'bcrypt';
+
+@Injectable()
+export class UsersService {
+  constructor(
+    @InjectRepository(User)
+    private userRepository: Repository<User>,
+    @InjectRepository(Login)
+    private loginRepository: Repository<Login>,
+  ) {}
+
+  async createUser(createUserDto: CreateUserDto): Promise<object> {
+    const { name, email, password } = createUserDto;
+    try {
+      // login token and userDetail are restore to different table
+      const user = new User();
+      user.name = name;
+      user.email = email;
+      const saveUser = await this.userRepository.save(user);
+      const userId = saveUser.id;
+
+      const userLogin = new Login();
+      userLogin.user_id = userId;
+      userLogin.name = name;
+      userLogin.token = await bcrypt.hash(password, 10);
+      const saveLogin = await this.loginRepository.save(userLogin);
+
+      return { saveUser, saveLoginId: saveLogin.id };
+    } catch (err) {
+      // if email already exists MySQL will present err.code '23505'
+      if (err.code === '23505') {
+        throw new HttpException('Email already exists', HttpStatus.BAD_REQUEST);
+      }
```

Comment:
Instead of checking the error code, checking what the instance of `err` is and do the corresponding handle is more standard.